ansible_collections/arista/avd/plugins/plugin_utils/eos_designs_facts.py

```diff
@@ -1624,7 +1616,96 @@ def overlay_routing_protocol_address_family(self):
         return overlay_routing_protocol_address_family
 
     @cached_property
-    def overlay_routing_protocol_peering_address(self):
+    def bgp(self):
+        """
+        Boolean telling if BGP Routing should be configured.
+        """
+        return (
+            self.underlay_router
+            and self.uplink_type == "p2p"
+            and (
+                self.underlay_routing_protocol == "ebgp"
+                or (
+                    self.overlay_routing_protocol in ["ebgp", "ibgp"]
+                    and (self.evpn_role in ["client", "server"] or self.mpls_overlay_role in ["client", "server"])
+                )
+            )
+        )
+
+    @cached_property
+    def evpn_encapsulation(self):
+        """
+        Toggle to set EVPN encapsulation based on fabric_evpn_encapsulation and node default_evpn_encapsulation.
+        """
+        return get(self._hostvars, "fabric_evpn_encapsulation", default=get(self._node_type_key_data, "default_evpn_encapsulation", default="vxlan"))
+
+    @cached_property
+    def underlay(self):
+        """
+        Returns a dictionary of underlay parameters to configure on the node.
+        """
+        if self.uplink_type != "p2p":
```

Comment:
Include `or not self.underlay_router` so you can remove that from all the other tests.

Reply:
Fixed.